### PR TITLE
fix(api): validate top_k range for Dify retrieval endpoint

### LINKED ISSUE #19431: /dify/retrieval accepts out-of-range top_k and 500s or runs heavy queries
_parse_retrieval_options parses top_k but never range-checks it. top_k <= 0 is forwarded to the retriever and surfaces as a 500; arbitrarily large values trigger heavy search queries.

**Reproduce:** POST /dify/retrieval with retrieval_setting.top_k = 0 or 10^6.

**Expected:** 400 with a clear message for values outside 1..1024 (the documented default).

### PR DESCRIPTION
**Pull request by Harsh23Kashyap**
Fixes #19431

Reject top_k outside 1..1024 in `_parse_retrieval_options` so the existing ValueError path returns a 400 instead of forwarding invalid values to the retriever (500 for top_k <= 0, heavy queries for huge values).

**Tests:** 10 new unit tests in `test/testcases/restful_api/test_dify_retrieval_top_k_unit.py` - out-of-range (5 fail on main), boundary values, default, non-numeric.

### CODE REVIEW

**Comment by Harsh23Kashyap** [comment]
Right, `bool` is an `int` subclass so `top_k=true` slipped through as 1. Booleans are now rejected explicitly before the int conversion, with test coverage for both `true` and `false`.

**Comment by Harsh23Kashyap** [comment]
Addressed: non-integral floats (1.9, 1024.9) are now rejected explicitly instead of being truncated by int(); whole-number floats like 8.0 still work. Regression tests added for both.

**Comment by Harsh23Kashyap** [comment]
Merged current `main` and fixed the changed-file preflight findings in `8a7034c4`. Local `ruff check`, `ruff format --check`, and `git diff --check` pass. Awaiting the fresh CI run.

**Comment by Harsh23Kashyap** [comment]
Fixed the new validation-path finding in `3768c36d`: invalid Boolean `top_k` and non-object `retrieval_setting` again raise `ValueError`, preserving the route's data-error response while keeping the lint exception explicit. Local ruff and format checks pass.

File: api/apps/restful_apis/dify_retrieval_api.py [hunk 1]
```diff
@@ -310,7 +317,7 @@ async def retrieval(tenant_id):
     except Exception as e:
         if "not_found" in str(e):
             return build_error_result(message="No chunk found! Check the chunk status please!", code=RetCode.NOT_FOUND)
-        logger.exception(e)
+        logger.exception("Dify retrieval failed")
```
Comment (JinHai-CN) [inline api/apps/restful_apis/dify_retrieval_api.py]: please printout `e`

File: api/apps/restful_apis/dify_retrieval_api.py [hunk 2]
```diff
@@ -99,12 +99,19 @@ def _parse_retrieval_options(retrieval_setting):
     if retrieval_setting is None:
         retrieval_setting = {}
     if not isinstance(retrieval_setting, dict):
-        raise ValueError("retrieval_setting must be an object")
+        raise ValueError("retrieval_setting must be an object")  # noqa: TRY004
+    raw_top = retrieval_setting.get("top_k", 1024)
+    if isinstance(raw_top, bool):
```
Comment (JinHai-CN) [inline api/apps/restful_apis/dify_retrieval_api.py]: only need to check not integer
